README.md

```diff
@@ -1,28 +1,157 @@
 ```
 
-## Troubleshooting
-Tracer sometimes fails to resolve Jaeger's address. In such a case run:
+To cleanup and remove everything, just run:
 ```bash
-sudo iptables -F
+oc delete all,template,daemonset -l jaeger-infra
+```
+
+## Production setup
+
+### TL;DR
+
+Install with:
+```bash
+oc login -u system:admin
+oc new-project jaeger-infra
+oc process -f production/template.yml | oc create -f -
+oc policy add-role-to-user cluster-admin developer -n jaeger-infra
+```
+
+Your Agent hostname is `jaeger-agent.jaeger-infra.svc.cluster.local`
+
+Clean up with:
+```bash
+oc delete all,template,daemonset -l jaeger-infra
+```
+
+### Detailed instructions
+
+You may choose to deploy the Jaeger instance on its own namespace or on an existing namespace,
+with the installation done by an admin or by an unprivileged user. In any case, you'll need your
+Tracer to be pointed to the Agent, which is available at a hostname following this pattern:
+
+```
+jaeger-agent.${NAMESPACE}.svc.cluster.local
+```
+
+For instance, when deployed on its own namespace `jaeger-infra`, this is the full hostname:
+
+```
+jaeger-agent.jaeger-infra.svc.cluster.local
+```
+
+#### Creating the deployment
+
+The `template.yml` file provides a quick start to deploy a production-ready instance of Jaeger on OpenShift.
+The template includes a `StatefulSet` for Cassandra, a `DaemonSet` for the Agent, a `Job` that creates the schema,
+as well as regular services for the individual components such as `Collector` and `Query`. Each one of those
+can be managed and scaled individually.
+
+#### Installing Jaeger on its own namespace as `admin`
+
+Ideally, the Jaeger components would run on its own namespace. For the recommended setup, these are the commands
+to be issued:
+
+```bash
+oc login -u system:admin
+oc new-project jaeger
+oc process -f template.yml | oc create -f -
+```
+
+To be able to manage the objects via the UI, add the appropriate permissions to your user. For development
+purposes, this can be executed to get the `developer` user to access the Jaeger namespace as `cluster-admin`:
+
+```bash
+oc login -u system:admin
+oc policy add-role-to-user cluster-admin developer -n jaeger
+```
+
+#### Installing Jaeger under an existing namespace as `developer`
+
+Jaeger can also be installed as `developer` (or any other user). As the `DaemonSet` runs at the node level,
+however, your user need to have such permission. It can be achieved by running this:
+
+```bash
+oc login -u system:admin
+oc create -f daemonset-admin.yml
+oadm policy add-role-to-user daemonset-admin developer
+```
+
+Once that is ready, it's only a matter of creating the components from the template:
+```bash
+oc process -f template.yml -p NAMESPACE=myproject | oc create -f -
+```
+
+#### Post install
+
+The Docker images should be pulled automatically from Docker Hub, and all other processes should also complete
```

Comment:
> The Docker images should be pulled automatically from Docker Hub

This is the way how templates usually work no? The most important thing from this long section is that some pods may fail for the first time.